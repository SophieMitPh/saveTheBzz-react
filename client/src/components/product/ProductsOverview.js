import React, { useState, useEffect } from "react";
import '../../App.css';

// SERVICES
import productService from './../../services/productService';

function ProductsOverview() {
	const [products, setproducts] = useState(null);

	useEffect(() => {
		if(!products) {
			getProducts();
		}
	})

	const getProducts = async () => {
		let res = await productService.getAll();
		console.log(res);
		setproducts(res);
	}

	const renderProduct = product => {
		return (
			<li key={product.name}>
				<div className="imgBorder">
					<a href={"/product/detail/" + product._id}>
						<img className="product-image" src={product.name + ".png"} alt="product"/>
						</a>
					<div className="belowpic">
						<h2 id="h2-item-name">{product.name}</h2>
						<a href="/"><img className="wishlistIcon" src="/wishList.png" alt="wishlist icon"/></a>
					</div>
				</div>
			</li>
		)
	};

	return (
		<div className="page">
			<h2 className="productTitle">
				Products
			</h2>
		{(products && products.length > 0) ? (
			products.map(product => renderProduct(product))
		) : (
			<p>No products found</p>
		)}
		</div>
	);
}

export default ProductsOverview;
