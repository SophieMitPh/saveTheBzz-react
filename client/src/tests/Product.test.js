import React from 'react';
import { shallow } from 'enzyme';
import ProductsOverview from './../components/product/ProductsOverview';
import { cleanup } from '@testing-library/react';
import axios from 'axios';

afterEach(cleanup)
jest.mock('axios');


describe('Product Overview', () => {
	it('should render correctly in "debug" mode', () => {
		const component = shallow(<ProductsOverview debug />);

		expect(component).toMatchSnapshot();
	});
});
