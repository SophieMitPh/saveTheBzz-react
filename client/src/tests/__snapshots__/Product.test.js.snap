// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Overview should render correctly in "debug" mode 1`] = `
<div
  className="page"
>
  <h2
    className="productTitle"
  >
    Products
  </h2>
  <p>
    No products found
  </p>
</div>
`;
